client/src/templates/Challenges/redux/completion-epic.js

```diff
@@ -97,7 +105,13 @@ function submitProject(type, state) {
     payload: challengeInfo
   };
   return postChallenge(update, username).pipe(
-    concat(of(updateSolutionFormValues({})))
+    switchMap(project => {
+      const { type, payload } = project;
+      if (type === challengeActionTypes.updateFailed) {
+        return of(updateFailed(payload));
+      }
+      return of(updateSolutionFormValues({}));
+    })
```

Comment:
I'm not sure about this.  The issue was, I thought, that action with `type == submitChallenge` was being dispatched twice, one of which would trigger the error flash and one of which would close the modal and navigate.

This solves the problem, but only by conditionally dropping the actions that're piped in.  Instead, should we simply only dispatch a single action?

Reply:
I am not sure. I thought it needed to dispatch one of those actions, in order for the chain to continue.

Reply:
I'm just not convinced two `submitChallenge` actions should be dispatched.

Reply:
@ahmadabdolsaheb am I missing something here?  What we're trying to achieve is:

User clicks submit -> client talks to server -> server responds with error -> user is warned

or

User clicks submit -> client talks to server -> server responds with success -> modal is closed -> navigate to next page

and what I'm worried about is that the action that's dispatched to trigger this *seems* to get dispatched twice.

Reply:
After chatting with @ahmadabdolsaheb we realised that ideally it should both ignore duplicate `submitChallenge`s *and* just dispatch a single action.  Reason being rapid submission is almost certainly a bug, so we should code defensively.  And, well, if a single click creates two submissions that's definitely a bug.